test(auth): add logout unit tests

MaksymChukhrai opened the pull request:
## Describe your changes

Enhanced logout functionality with session cleanup and fixed OAuth duplicate email bug. Added 13 unit tests.

## Issue ticket code

- [ALRAP-280](https://lizbethwangari2018.atlassian.net/browse/ALRAP-280)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Imports should come in a specific order
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants (N/A)
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests (13 unit tests ✅)
- [x] Check commit messages meet conventional commit format

### Backend

- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [x] Unit tests written (13 tests passing)
- [x] use ConfigService instead of process.env
- [x] use transactions for data mutations
- [x] use @index decorator for frequently requested data

## Changes Made

### Backend
- Fixed OAuth duplicate email bug (Google ↔ LinkedIn)
- Enhanced logout with database transaction
- Session destruction + connect.sid cleanup
- Comprehensive logging with Logger service
- Global JwtAuthGuard with @Public() decorator

### Frontend Integration
- Updated logout to call backend API
- Fixed endpoint URL (/api/auth/logout)

### Unit tests
```
# Run logout-related tests only
npm run test -- --testNamePattern="logout"

# Or run specific test files
npm run test -- auth.service.spec.ts
npm run test -- auth.controller.spec.ts
```

## Test Results
```bash
Test Suites: 2 passed, 2 total
Tests:       13 passed, 13 total
Time:        7.263s
```

MaksymChukhrai committed:
test(auth): add logout unit tests

- Added 13 unit tests for logout functionality
- AuthService: 7 tests (transaction, error handling, token cleanup)
- AuthController: 6 tests (session, cookies, error handling)
- Updated jest.config.js and tsconfig.json for test support
- All tests passing ✅

ALRTAP-280

lyudad commented on postman/auth-tests.postman_collection.json:
why postman needed? there is swagger

MaksymChukhrai replied:
Postman folder deleted

lyudad commented on src/modules/auth/auth.service.ts:
```diff
@@ -33,27 +33,44 @@ export class AuthService {
+   *
+   * @param profile - OAuth profile from provider
+   * @returns Authentication response with tokens
+   */
```
comments not needed

MaksymChukhrai replied:
deleted

lyudad commented on src/modules/auth/auth.service.ts:
```diff
@@ -33,27 +33,44 @@ export class AuthService {
       });
 
       if (!user) {
+        // Create new user
```
its obvious from the code

MaksymChukhrai replied:
deleted

lyudad commented on src/modules/auth/auth.service.ts:
```diff
@@ -33,27 +33,44 @@ export class AuthService {
+          `New user created via ${profile.provider}: ${profile.email}`,
+        );
       } else {
+        // Update existing user with new provider info
```
its obvious comment not needed

MaksymChukhrai replied:
deleted

MaksymChukhrai committed:
refactor: remove Postman collection and clean up auth.service comments

- Remove postman/auth-tests.postman_collection.json (use Swagger instead)
- Remove unnecessary JSDoc comments from auth.service.ts
- Remove inline comment about updating existing user
